File: problem_004.pl
#!/usr/bin/env perl

print "Project Euler - Problem 4:\n";
print "Find the largest palindrome made from the product of two 3-digit numbers.\n\n";

$result = 0;

for ($a=100 ; $a<1000 ; $a++) {
    for ($b=$a ; $b<1000 ; $b++) {
        $product = $a * $b;
        $tcudorp = 0;
        while ($product > 0) {
            $tcudorp *= 10;
            $tcudorp += $product % 10;
            $product = int($product/10);
        }
        $product = $a * $b;
        #print $product, " : ", $tcudorp, "\n"; #DEBUG
        if ($product == $tcudorp) {
            if ($result < $product) {
                $result = $product;
            }
        }
    }
}

print "Largest product: ", $result, "\n";

File: problem_004.lua
#!/usr/bin/env lua

print("Project Euler - Problem 4:")
print("Find the largest palindrome made from the product of two 3-digit numbers.\n")

result = 0

for a = 999, 100, -1 do
   for b = a, 100, -1 do
      product = a * b
      tcudorp = 0
      while (product > 0) do
         tcudorp = math.floor(tcudorp * 10)
         tcudorp = tcudorp + (product % 10)
         product = math.floor(product / 10)
      end
      product = a * b
      --print (product.." "..tcudorp) --DEBUG
      if ( product == tcudorp ) then
         if result < product then
            result = product
         end
      end
   end
end

print("Largest product: "..result)
